app/cdap/components/DirectiveInput/Components/InputPanel/index.tsx

```diff
@@ -0,0 +1,228 @@
+    mousetrap.bind('enter', handleEnterKey);
+    mousetrap.bind('tab', handleTabKey);
+
+    // unbind a single keyboard event or an array of keyboard events.
+    return () => {
+      mousetrap.unbind('up');
+      mousetrap.unbind('down');
+      mousetrap.unbind('enter');
+      mousetrap.unbind('tab');
+      eventEmitter.off(globalEvents.DIRECTIVEUPLOAD, getUsage);
+    };
+  });
+
+  const handleUpArrow = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    if (selectedIndex !== 0) {
+      setSelectedIndex(selectedIndex - 1);
+    }
+  };
+
+  const handleDownArrow = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    if (selectedIndex !== searchResults.length - 1) {
+      setSelectedIndex(selectedIndex + 1);
+    }
+  };
+
+  const handleEnterKey = () => {
+    if (inputText.length > 0) {
+      if (searchResults[selectedIndex]) {
+        handleListItemClick(searchResults[selectedIndex]);
+      } else {
+        onSearchItemClicked(inputText);
+      }
+    }
+  };
+
+  const handleTabKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    if (inputText.length === 0 || inputText.split(' ').length !== 1) {
+      return;
+    }
+    handleEnterKey();
+  };
+
+  useEffect(() => {
+    if (directiveSelected) {
+      setFuse(new Fuse(columnNamesList, { ...defaultFuseOptions, keys: ['label'] }));
+    }
+    searchMatch(inputBoxValue);
+    setInputText(inputBoxValue);
+  }, [inputBoxValue]);
+
+  const searchMatch = (searchString: string) => {
+    let searchList = [];
+    const spaceIndex: boolean = searchString.includes(' ');
+    if (fuse && searchString.length > 0) {
+      if (!directiveSelected) {
+        searchList = fuse
+          .search(searchString)
+          .slice(0, 3)
+          .map((searchItem) => {
+            searchItem.uniqueId = uuidV4();
+            return searchItem;
+          });
+        reverse(searchList);
+      } else {
+        searchList = fuse.search(searchString.split(':')[1]).map((searchItem) => {
+          searchItem.uniqueId = uuidV4();
+          return searchItem;
+        });
+        reverse(searchList);
+      }
+    }
+    setSearchResults(searchList);
+    setInputText(searchString);
+    setSelectedIndex(searchList.length - 1);
+    if (!directiveSelected) {
+      getDirectiveSyntax(searchList, spaceIndex);
+    }
+  };
+
+  const handleListItemClick = (listItem) => {
+    if (!directiveSelected) {
+      onSearchItemClicked(listItem.item.directive);
+      getDirectiveSyntax([listItem], true);
+    } else {
+      const splitData = inputText.split(/(?=[:])|(?<=[:])/g);
+      const eventObject: Record<string, IObject> = {
+        target: { value: `${splitData[0]}${splitData[1]}${listItem.item.label}` },
+      };
+      setInputText(`${splitData[0]}${splitData[1]}${listItem.item.label}`);
+      onSearchItemClicked(eventObject.target.value);
+      onColumnSelected(true);
```

Comment:
should this be onColumnSelection

plz check

Reply:
Done